src/transformers/generation_utils.py

```diff
@@ -1976,6 +1976,18 @@ def sample(
         >>> tokenizer.batch_decode(outputs, skip_special_tokens=True)
         ['Today is a beautiful day, and a wonderful day.\n\nI was lucky enough to meet the']
         ```"""
+        if self.device.type != input_ids.device.type:
+            warnings.warn(
+                "You are calling .generate() with the `input_ids` being on a device type different"
+                f" than your model's device. `input_ids` is on {input_ids.device.type}, whereas the model"
+                f" is on {self.device.type}. You may experience unexpected behaviors."
+                " Please make sure that you have put `input_ids` to the"
+                " correct GPU by calling for example input_ids = input_ids.to('cuda') before"
+                " running `.generate()`.",
+                UserWarning,
+            )
+            first_device = list(self.parameters())[0].device  # retrieve the first device for accelerate compatibility
```

Comment:
Why not `self.device` ?

Reply:
Because I was unsure about how `self.device` is handled in `generate`, what device is used if the model is loaded across multiple GPUs?

Reply:
Just tried it with a model sharded across 2 GPUs and `self.device` returns the first device, so we can probably safely use this instead

Reply:
Fixed in e9b49eb3b3bd85d668df5fa4cb8b9f9c576fcb54 !